index.js

```diff
@@ -265,46 +281,50 @@ const awsProcessing = (dist, cb) => {
             config.awsami = body.imageId;
             config.awstype = body.instanceType;
         }
-        return macaddress.all(cb);
+        return macaddress.all((error, all) => {
+            if (error) {
+                log(`Error in Getting MacAddress: ${error}`);
+            }
+            return cb(null, error ? {} : all);
+        });
     });
 };
 
 if ((os.platform() === 'win32' && require('is-administrator')()) || process.getuid() <= 0) {
-    async.waterfall([checkFileExistence, mainExecution, awsProcessing]
-        , (error, all) => {
-            if (error) {
-                console.error(error);
-                process.exit();
-            }
-            if (all) {
-                var ifaces = Object.keys(all);
-                for (var i = 0; i < ifaces.length; i++) {
-                    if (
-                        all[ifaces[i]].ipv4 && (
-                            all[ifaces[i]].ipv4.indexOf('10.') === 0 ||
+    async.waterfall([
+        checkFileExistence
```

Comment:
actually, why have this encapsulate 2 different async calls instead of having those be expanded here inside the waterfall?

Reply:
Done!